graph-commons/src/main/scala/io/renku/data/ErrorMessage.scala

```diff
@@ -18,36 +18,59 @@
 
 package io.renku.data
 
+import cats.Show
 import eu.timepit.refined.api.{RefType, Refined}
 import eu.timepit.refined.collection.NonEmpty
+import io.circe.Json
 
 import java.io.{PrintWriter, StringWriter}
 
+sealed trait ErrorMessage {
+  type T
+  val value: T
+  def show: String
+}
+
 object ErrorMessage {
 
-  type ErrorMessage = String Refined NonEmpty
+  final case class StringMessage(value: String Refined NonEmpty) extends ErrorMessage {
```

Comment:
Or maybe remove that string message and always use a json message? it could be instantiated by a string to mimic the string message, but would still be json. I think we don't encode these really differently, so if we have a JsonMessage anyways, we could also use it for everything.

Reply:
We could but sometimes we use `ErrorMessage` factory to create a break and line feed chars free version of String that can be put as a value of tiny types that are stored in Postgres (e.g. `EventMessage.apply`). Sure this logic can be a separate tool, too but it's somehow easier to use/find under an `ErrorMessage` name.
The other usage of the class is to create an error message once and then do both: log it and encode it into the HTTP response.